### PR TITLE
MAP-86

### PR DESCRIPTION
**Pull request by hkf-tech**

### CODE REVIEW

File: src/Approvals/SFA.DAS.Approvals.Api/Controllers/OverlappingTrainingDateRequestController.cs [hunk 1]
```diff
@@ -41,5 +43,21 @@ public async Task<IActionResult> Validate([FromBody] ValidateDraftApprenticeship
 
             return Ok(result);
         }
+
+        /// TODO: Unit Test on this
```
Comment (cofaulco) [inline src/Approvals/SFA.DAS.Approvals.Api/Controllers/OverlappingTrainingDateRequestController.cs:47]: TODO. Has this been done?

File: src/Approvals/SFA.DAS.Approvals/Application/OverlappingTrainingDateRequest/Queries/ValidateUlnOverlapOnStartDateQueryHandler.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+using SFA.DAS.Approvals.InnerApi.CommitmentsV2Api.Responses;
+using SFA.DAS.SharedOuterApi.Configuration;
+using SFA.DAS.SharedOuterApi.Extensions;
+using SFA.DAS.SharedOuterApi.Interfaces;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.Approvals.Application.OverlappingTrainingDateRequest.Queries
+{
+    /// TODO : Unit test
```
Comment (cofaulco) [inline src/Approvals/SFA.DAS.Approvals/Application/OverlappingTrainingDateRequest/Queries/ValidateUlnOverlapOnStartDateQueryHandler.cs:12]: TODO has this been done?

File: src/Approvals/SFA.DAS.Approvals/InnerApi/CommitmentsV2Api/Requests/ValidateUlnOverlapOnStartDateQueryRequest.cs [hunk 3]
```diff
@@ -0,0 +1,23 @@
+﻿using SFA.DAS.SharedOuterApi.Interfaces;
+
+namespace SFA.DAS.Approvals.InnerApi.CommitmentsV2Api.Requests
+{
+    /// TODO : Unit Test
```
Comment (cofaulco) [inline src/Approvals/SFA.DAS.Approvals/InnerApi/CommitmentsV2Api/Requests/ValidateUlnOverlapOnStartDateQueryRequest.cs:5]: TODO ? Has this been done?

**Review by cofaulco — CHANGES_REQUESTED** [review]
Missing unit tests mentioned in TODO comments

File: src/Approvals/SFA.DAS.Approvals.Api/Controllers/OverlappingTrainingDateRequestController.cs (hunk 1 above)
Comment (kola-samuel) [reply src/Approvals/SFA.DAS.Approvals.Api/Controllers/OverlappingTrainingDateRequestController.cs]: Done

File: src/Approvals/SFA.DAS.Approvals/InnerApi/CommitmentsV2Api/Requests/ValidateUlnOverlapOnStartDateQueryRequest.cs (hunk 3 above)
Comment (kola-samuel) [reply src/Approvals/SFA.DAS.Approvals/InnerApi/CommitmentsV2Api/Requests/ValidateUlnOverlapOnStartDateQueryRequest.cs:5]: Done

Comment (cofaulco) [reply src/Approvals/SFA.DAS.Approvals/InnerApi/CommitmentsV2Api/Requests/ValidateUlnOverlapOnStartDateQueryRequest.cs:5]: the TODO comment hasn't been removed

Comment (cofaulco) [inline src/Approvals/SFA.DAS.Approvals/InnerApi/CommitmentsV2Api/Requests/ValidateUlnOverlapOnStartDateQueryRequest.cs:5]: this has been marked as resolved, but the todo still exists

**Review by cofaulco — CHANGES_REQUESTED** [review]

File: src/Approvals/SFA.DAS.Approvals/InnerApi/CommitmentsV2Api/Requests/ValidateUlnOverlapOnStartDateQueryRequest.cs (hunk 3 above)
Comment (kola-samuel) [reply src/Approvals/SFA.DAS.Approvals/InnerApi/CommitmentsV2Api/Requests/ValidateUlnOverlapOnStartDateQueryRequest.cs]: Marked the wrong one test now added thanks

**Review by cofaulco — APPROVED** [review]

**Pull request merged**